plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -189,19 +194,21 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 	}
 	processNode := func(i int) {
 		nodeInfo := nodeNameToInfo[allNodeNames[i]]
-		if hasAffinityConstraints || hasAntiAffinityConstraints {
-			// We need to process all the nodes.
-			for _, existingPod := range nodeInfo.Pods() {
-				if err := processPod(existingPod); err != nil {
-					pm.setError(err)
+		if nodeInfo.Node() != nil {
```

Comment:
I don't think this check is necessary.

Reply:
@bsalamat  Can you share more your thoughts?

Reply:
nodeInfo.Node() can be nil when scheduler has received a delete for the node, but scheduler has not yet received delete events for all the pods on the node. In this kind of scenario it may be ok to consider those pods for affinity/anti-affinity. That said, I don't see a problem with adding this check.

Reply:
Can you check my following comments about this issue?
It is used to fix another scenario.